resolver.go

```diff
@@ -136,7 +143,13 @@ func (r *resolver) Stop() {
 }
 
 func (r *resolver) SetExtServers(dns []string) {
-	r.extDNS = dns
+	l := len(dns)
+	if l > maxExtDNS {
```

Comment:
Forget what I suggested. I was assuming dns is always maxExtDNS but it could be less in which case this will panic.